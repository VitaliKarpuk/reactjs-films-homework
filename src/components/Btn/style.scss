.btn__watch,
.btn__info {
  width: 199px;
  height: 60px;
  border-radius: 6px;
  margin: 5px;
  font-size: 28px;
  letter-spacing: 1px;
  line-height: 44px;
  color: #ffffff;
  text-align: center;
  margin-right: 20px;
}

.btn__watch {
  background-color: #0aaee4;
  border: none;

  &:hover {
    cursor: pointer;
  }
}

.btn__info {
  background-color: transparent;
  border: 2px solid #ffffff;

  &:hover {
    cursor: pointer;
  }
}
