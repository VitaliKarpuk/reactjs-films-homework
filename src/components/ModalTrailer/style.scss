.modal__trailer{
  position: fixed;
  min-width: 100vw;
  height: 100vh;
  background-color: #8a8a8a9e;
  top: 0;
  left: 0;
  z-index: 7;

  .modali__trailer_window{
    width: 80%;
    height: 70vh;
    margin: 0 auto;
    margin-top: 15vh;
    background-color: rgb(94, 94, 94);
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    div{
      height: 90%;
      display: flex;
      align-items: center;
      justify-content: center;
      background-color: aliceblue;
    }
    iframe{
      width: 100%;
      height: 100%;
      border: none;
    }
  }
}