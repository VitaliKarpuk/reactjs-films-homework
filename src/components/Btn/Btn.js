import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';

const Btn = ({ title, className, onClick }) => {

  return (
    <button className={className} onClick={onClick}>{title}</button>
  )
}

Btn.propTypes = {
  title: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func
}

export default Btn;