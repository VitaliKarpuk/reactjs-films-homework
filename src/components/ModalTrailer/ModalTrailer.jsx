import React from "react";
import PropTypes from "prop-types";

import "./style.scss";

const ModalTrailer = ({ className, handleCloseTrailer, trailer }) => {
  return (
    <div className="modal__trailer">
      <div className="modali__trailer_window">
        <button className={className} onClick={handleCloseTrailer}>
          close
        </button>
        <div>
          <iframe src={`https://www.youtube.com/embed/${trailer}`}></iframe>
        </div>
      </div>
    </div>
  );
};

ModalTrailer.propTypes = {
  handleCloseTrailer: PropTypes.func,
  className: PropTypes.string,
  trailer: PropTypes.string
};

ModalTrailer.defaultProps = {
  handleCloseTrailer: () => {},
  className: "",
  trailer: ""
};
export default ModalTrailer;
