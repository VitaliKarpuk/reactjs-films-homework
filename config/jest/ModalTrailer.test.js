/* eslint-disable no-undef */
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow';
import ModalTrailer from '../../src/components/ModalTrailer/ModalTrailer';
import Btn from '../../src/components/Btn/Btn';

describe('Test ModalTrailer component', () => {
  test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<ModalTrailer />);
    renderer.getRenderOutput();
    expect(result.props.children.length).toBe(4);
    expect(result.props.children[0].type).toBe('div');
    expect(result.props.children[1].type).toBe('h3');
    expect(result.props.children[2]).toMatchObject(<Btn />)
    result.props.children[0].props.onClick();
    result.props.children[2].props.onClick();
    result.props.children[3].props.onOk();
    result.props.children[3].props.onCancel();

    expect(result).toMatchSnapshot();
  })
})
