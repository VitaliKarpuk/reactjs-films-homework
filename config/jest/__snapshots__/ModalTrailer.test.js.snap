// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ModalTrailer component renders correctly 1`] = `
<div
  className="modal__trailer"
>
  <div
    className="modali__trailer_window"
  >
    <button
      className=""
      onClick={[Function]}
    >
      close
    </button>
    <div>
      <iframe
        src="https://www.youtube.com/embed/"
      />
    </div>
  </div>
</div>
`;
